project_2/include/packet.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <cstdint>
+#include <ostream>
+
+/**
+ * @brief Network packet abstraction
+ * 
+ */
+class PacketHeader {
+private:
+    std::uint64_t source_id;
+    std::uint64_t destination_id;
+    std::uint64_t sending_time;
+    std::uint32_t packet_index;
+    std::uint32_t size;
+
+    // 0 bit - is credit packet
+    // 2 bit - initializing connection or not
+    std::uint8_t flags = 0;
+public:
+    PacketHeader() = default;
+    /**
+     * @brief Construct a new Packet Header object
+     * 
+     * @param source_id packet source
+     * @param destination_id packet destination
+     * @param sending_time time of sending packet 
+     * @param packet_index unique packet identifier (within network element, that generated it)
+     * @param size packet size
+     */
+    PacketHeader(std::uint64_t source_id, std::uint64_t destination_id, std::uint64_t sending_time, std::uint32_t packet_index, std::uint32_t size);
+
+    std::uint32_t GetPackageIndex() const;
+    std::uint64_t GetDestinationID() const;
+    std::uint64_t GetSendingTime() const;
+    std::uint64_t GetSourceID() const;
+    std::uint32_t GetSize() const;
+    std::uint8_t GetFlag(std::uint8_t bit) const;
```

Comment:
Could you, please, rewrite the code without using getters and setters which obscure the meaning and add too much verbosity. See the opinion on this here https://youtu.be/uHSLHvWFkto?t=4380 discussion and further discussion https://stackoverflow.com/questions/32625274/public-variables-bad-practice-vs-getters-and-setters-functions. It is better to write the most simple and working code first and then gradually make it more complex step by step.